orchagent/srv6orch.cpp

```diff
@@ -0,0 +1,654 @@
+            if (status != SAI_STATUS_SUCCESS)
+            {
+                SWSS_LOG_ERROR("Failed to remove SRV6 tunnel object for source %s", srv6_source.c_str());
+                return false;
+            }
+            srv6_tunnel_table_.erase(srv6_source);
+        }
+        else
+        {
+            SWSS_LOG_INFO("Nexthops referencing this tunnel object %s: %zu", srv6_source.c_str(),tunnel_nhs);
+        }
+    }
+    return true;
+}
+
+bool Srv6Orch::createSrv6Nexthop(const NextHopKey &nh)
+{
+    SWSS_LOG_ENTER();
+    string srv6_segment = nh.srv6_segment;
+    string srv6_source = nh.srv6_source;
+
+    if (srv6NexthopExists(nh))
+    {
+        SWSS_LOG_INFO("SRV6 nexthop already created for %s", nh.to_string(false,true).c_str());
+        return true;
+    }
+    sai_object_id_t srv6_object_id = sid_table_[srv6_segment].sid_object_id;
+    sai_object_id_t srv6_tunnel_id = srv6_tunnel_table_[srv6_source].tunnel_object_id;
+
+    if (srv6_object_id == SAI_NULL_OBJECT_ID)
+    {
+        SWSS_LOG_ERROR("segment object doesn't exist for segment %s", srv6_segment.c_str());
+        return false;
+    }
+
+    if (srv6_tunnel_id == SAI_NULL_OBJECT_ID)
+    {
+        SWSS_LOG_ERROR("tunnel object doesn't exist for source %s", srv6_source.c_str());
+        return false;
+    }
+    SWSS_LOG_INFO("Create srv6 nh for tunnel src %s with seg %s", srv6_source.c_str(), srv6_segment.c_str());
+    vector<sai_attribute_t> nh_attrs;
+    sai_object_id_t nexthop_id;
+    sai_attribute_t attr;
+    sai_status_t status;
+
+    attr.id = SAI_NEXT_HOP_ATTR_TYPE;
+    attr.value.s32 = SAI_NEXT_HOP_TYPE_SRV6_SIDLIST;
+    nh_attrs.push_back(attr);
+
+    attr.id = SAI_NEXT_HOP_ATTR_SRV6_SIDLIST_ID;
+    attr.value.oid = srv6_object_id;
+    nh_attrs.push_back(attr);
+
+    attr.id = SAI_NEXT_HOP_ATTR_TUNNEL_ID;
+    attr.value.oid = srv6_tunnel_id;
+    nh_attrs.push_back(attr);
+
+    status = sai_next_hop_api->create_next_hop(&nexthop_id, gSwitchId,
+                                                (uint32_t)nh_attrs.size(),
+                                                nh_attrs.data());
+    if (status != SAI_STATUS_SUCCESS)
+    {
+        SWSS_LOG_ERROR("FAiled to create srv6 nexthop for %s", nh.to_string(false,true).c_str());
+        return false;
+    }
+    m_neighOrch->updateSrv6Nexthop(nh, nexthop_id);
+    srv6_nexthop_table_[nh] = nexthop_id;
+    sid_table_[srv6_segment].nexthops.insert(nh);
+    srv6TunnelUpdateNexthops(srv6_source, nh, true);
+    return true;
+}
+
+bool Srv6Orch::srv6Nexthops(const NextHopGroupKey &nhgKey, sai_object_id_t &nexthop_id)
+{
+    SWSS_LOG_ENTER();
+    set<NextHopKey> nexthops = nhgKey.getNextHops();
+    string srv6_source;
+    string srv6_segment;
+
+    for (auto nh : nexthops)
+    {
+        srv6_source = nh.srv6_source;
+        if (!createSrv6Tunnel(srv6_source))
+        {
+            SWSS_LOG_ERROR("Failed to create tunnel for source %s", srv6_source.c_str());
```

Comment:
You want to continue create nexthop if Tunnel failed?

Reply:
Thanks..done